### ISSUE TITLE
New admin-area task page

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-web#480]
As of the upcoming 2.0.5 version, fractal-server has a new superuser-only `GET /admin/v2/task/` endpoint (thanks @ychiucco!). This is a first version of this endpoint, which we can still refine in the future. Its first use case is to help with issues like https://github.com/fractal-analytics-platform/fractal-server/issues/1457.

---

It's a GET request (without request body), with a bunch of optional query parameters to filter the task list:
```python
    id: Optional[int] = None,   # if set, only find the single task with this ID
    source: Optional[str] = None,    # if set, only find tasks with source that contains (case-insensitive) this string
    version: Optional[str] = None,   # if set, only find tasks with this version
    name: Optional[str] = None,    # if set, only find tasks with name that contains (case-insensitive) this string
    owner: Optional[str] = None,   # if set, only find tasks with this owner
    kind: Optional[Literal["common", "users"]] = None,    # if set to common, only find tasks with owner=None. if set to users, only find tasks with non-None owner
    max_number_of_results: int = 25,  # this is provisionally here, to avoid creating very large response bodies. After playing a bit with a real-life database we can refine this default value, or even remove this query parameter.
```

---

On this page, the admin can set some query filters for tasks, and get a list of enriched task objects. Each one of this objects has some basic task properties (id, name, commands, ..) and a list of relationships (including the list of workflows-projects-users associated to it).

The specific response-body models are available in https://fractal-analytics-platform.github.io/fractal-server/openapi/openapi.json (see also https://github.com/fractal-analytics-platform/fractal-server/blob/main/fractal_server/app/routes/admin/v2.py), and an example response body (https://github.com/fractal-analytics-platform/fractal-server/pull/1465#issuecomment-2097738828) looks like
```json
[
  {
    "task": {
      "id": 1,
      "name": "Foo",
      "type": "compound",
      "command_non_parallel": "cmd_non_parallel",
      "command_parallel": "cmd_parallel",
      "source": "xxx",
      "owner": "alice",
      "version": "0"
    },
    "relationships": [
      {
        "workflow_id": 1,
        "workflow_name": "My Workflow",
        "project_id": 1,
        "project_name": "project",
        "project_users": [
          {
            "id": 1,
            "email": "8374848@example.org"
          }
        ]
      },
      {
        "workflow_id": 2,
        "workflow_name": "My Workflow",
        "project_id": 1,
        "project_name": "project",
        "project_users": [
          {
            "id": 1,
            "email": "8374848@example.org"
          }
        ]
      }
    ]
  },
  {
    "task": {
      "id": 2,
      "name": "abcdef",
      "type": "compound",
      "command_non_parallel": "cmd_non_parallel",
      "command_parallel": "cmd_parallel",
      "source": "yyy",
      "owner": "bob",
      "version": "0"
    },
    "relationships": [
      {
        "workflow_id": 2,
        "workflow_name": "My Workflow",
        "project_id": 1,
        "project_name": "project",
        "project_users": [
          {
            "id": 1,
            "email": "8374848@example.org"
          }
        ]
      }
    ]
  },
  {
    "task": {
      "id": 3,
      "name": "task3",
      "type": "compound",
      "command_non_parallel": "cmd_non_parallel",
      "command_parallel": "cmd_parallel",
      "source": "source3",
      "owner": null,
      "version": "3"
    },
    "relationships": []
  }
]
```

**Comment by tcompa** [comment]
(fractal-server v2.0.5 is now available on PyPI)

**Comment by zonia3000** [comment]
This is a first version of the tasks admin page.

To avoid having a too big table I've displayed only some data in the table and I added an "Info" button that opens a modal showing the complete information.

I'm not sure if we have to display the source column, since its content is usually quite long, but I think it would be a bit weird to have it in the filters and not in the table.

![image](https://github.com/fractal-analytics-platform/fractal-web/assets/5459334/f066f484-78db-4acd-9749-6c7f683659b2)

The info modal displays the task properties and task relationships in separate accordions:

![image](https://github.com/fractal-analytics-platform/fractal-web/assets/5459334/d5046940-1456-4b5c-9d6e-fa316f8bdb6e)

![image](https://github.com/fractal-analytics-platform/fractal-web/assets/5459334/ae6bd0ae-0f64-4412-b10a-838e3a8de08f)

![image](https://github.com/fractal-analytics-platform/fractal-web/assets/5459334/c5d4d1ff-a42c-4f51-b734-bc1992160a68)

I think we can use a dropdown to select the owner filter. I'm currently filling the list using the usernames, falling back to the email if the username is null. Is this approach correct?

I imagined that a Delete button could be also useful but I'm not sure if we want to implement a mechanism that first delete all the references and then deletes the task.

I've also started to see this error:

![image](https://github.com/fractal-analytics-platform/fractal-web/assets/5459334/19eccfd4-222e-4f9a-9cdb-c682a6f57b1e)

How we should handle this case? Should I add the `max_number_of_results` in the form and leave the error as it is?

**Comment by tcompa** [comment]
> I think we can use a dropdown to select the owner filter. I'm currently filling the list using the usernames, falling back to the email if the username is null. Is this approach correct?

The correct fallback is slightly different.
If `username` is set, this is the value to use. If not, we should look for the `slurm_user` property of the user. If not even that one is set, then there is no additional fallback.

**Comment by tcompa** [comment]
> How we should handle this case? Should I add the max_number_of_results in the form and leave the error as it is?

We included this option provisionally, until we test this page/endpoint a bit further.
It's not urgent to handle this error in any special way, but if you can also expose this value in the form maybe it's the easiest solution for the moment. I think we'll revisit this feature in the future, anyway.

**Comment by tcompa** [comment]
Addendum to our ongoing discussion

The "info" messages could look like this:

- name: Only include a task if its `name` contains this value (case insensitive).
- kind: If `common`, only include tasks with null `owner`. If `users`, only include tasks with non-null `owner`. 
- owner: Only include a task if its `owner` matches this value.
- source: Only include a task if its `source` contains this value (case insensitive).
- version: Only include a task if its `version` matches this value.
- id: Only include the task with this `id` (if any).
- max_number_of_results: Upper limit on the number of tasks in the response.

**Comment by tcompa** [comment]
now closed with #484
